File: frontend/types/issue.ts
export type IssueSeverity = "LOW" | "MEDIUM" | "HIGH" | "CRITICAL";
export type IssueStatus = "OPEN" | "IN_PROGRESS" | "RESOLVED";

export interface Issue {
  id: number;
  project_id: number;
  /** Null for problems logged at project level without an owning task. */
  task_id: number | null;
  reported_by: number;
  title: string;
  description: string;
  severity: IssueSeverity;
  status: IssueStatus;
  suggested_solution: string | null;
  suggested_solution_is_ai?: boolean;
  created_at: string;
  resolved_at: string | null;
  task?: { id: number; task_name: string } | null;
  reporter?: { id: number; name: string; username: string } | null;
}

export const OPEN_ISSUE_STATUSES: IssueStatus[] = ["OPEN", "IN_PROGRESS"];

File: backend/app/models/issue.py
"""Issue ORM model."""

from __future__ import annotations

import enum
from datetime import datetime

from sqlalchemy import DateTime, Enum, ForeignKey, String, Text
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.core.database import Base
from app.models.base import TimestampMixin


class IssueSeverity(enum.StrEnum):
    LOW = "LOW"
    MEDIUM = "MEDIUM"
    HIGH = "HIGH"
    CRITICAL = "CRITICAL"


class IssueStatus(enum.StrEnum):
    OPEN = "OPEN"
    IN_PROGRESS = "IN_PROGRESS"
    RESOLVED = "RESOLVED"


OPEN_ISSUE_STATUSES = (IssueStatus.OPEN, IssueStatus.IN_PROGRESS)


class Issue(TimestampMixin, Base):
    __tablename__ = "issues"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    project_id: Mapped[int] = mapped_column(
        ForeignKey("projects.id", ondelete="CASCADE"),
        index=True,
        nullable=False,
    )
    # Optional: issues raised by the AI progress analyzer always have a task,
    # but manually logged project-level problems may have none.
    task_id: Mapped[int | None] = mapped_column(
        ForeignKey("tasks.id", ondelete="CASCADE"),
        index=True,
        nullable=True,
    )
    reported_by: Mapped[int] = mapped_column(
        ForeignKey("users.id", ondelete="RESTRICT"),
        index=True,
        nullable=False,
    )
    title: Mapped[str] = mapped_column(String(300), nullable=False)
    description: Mapped[str] = mapped_column(Text, nullable=False)
    severity: Mapped[IssueSeverity] = mapped_column(
        Enum(IssueSeverity, name="issue_severity", native_enum=False, length=16),
        nullable=False,
        default=IssueSeverity.MEDIUM,
    )
    status: Mapped[IssueStatus] = mapped_column(
        Enum(IssueStatus, name="issue_status", native_enum=False, length=16),
        nullable=False,
        default=IssueStatus.OPEN,
    )
    suggested_solution: Mapped[str | None] = mapped_column(Text, nullable=True)
    resolved_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), nullable=True)

    project = relationship("Project")
    task = relationship("Task")
    reporter = relationship("User", foreign_keys=[reported_by])
